File: 0229-majority-element-ii/0229-majority-element-ii.java
class Solution {
    public List<Integer> majorityElement(int[] nums) {
        Arrays.sort(nums);
        
        List<Integer> ans = new ArrayList<>();
        
        int i=0,freq=1,n=nums.length/3;
        
        while(i < nums.length){
            while(i+1 < nums.length && nums[i]==nums[i+1]){
                freq++;
                i++;
            }
            if(freq > n){
                ans.add(nums[i]);
            }
            
            freq=1;
            i++;
        }
        
        return ans;
    }
}

File: 0229-majority-element-ii/0229-majority-element-ii.cpp
class Solution {
public:
    vector<int> majorityElement(vector<int>& nums) {
        unordered_map<int,int>mp;
        for(auto it:nums){
            mp[it]++;
        }
        
        int n= nums.size()/3;
        vector<int>ans;
        for(auto it:mp){
            if(it.second > n){
                ans.push_back(it.first);
            }
        }
        
        return ans;
    }
};